pkg/operator/controller/status/controller.go

```diff
@@ -243,7 +251,7 @@ func (r *reconciler) Reconcile(ctx context.Context, request reconcile.Request) (
 			r.config.IngressControllerImage,
 			r.config.CanaryImage,
 		),
-		computeOperatorDegradedCondition(state.IngressControllers),
+		computeOperatorDegradedCondition(state),
```

Comment:
I would prefer to see it like this:
```suggestion
		computeGatewayAPICRDsDegradedCondition(state),		 
                computeOperatorDegradedCondition(state.IngressControllers),
```
See https://github.com/openshift/cluster-ingress-operator/pull/1205/files#r2027730400

Reply:
What you proposed erases `GatewayAPICRDsDegraded` reason and the corresponding message if it's set in `computeGatewayAPICRDsDegradedCondition` because `mergeConditions` function [overrides them](https://github.com/openshift/cluster-ingress-operator/blob/716f3c554d527e264ab9655db08ae1dd4e653fb9/pkg/operator/controller/status/controller.go#L687-L689) for the same condition type (`Degraded` in our case). What we want is to join 2 degraded conditions into a single one.

Reply:
As I mentioned on slack, the retries are for Reconciles.  This new Degraded condition for when there are unmanaged CRDs isn't being re-examined (or retried) on Reconcile anyway.  The logic is a real departure from the way we normally do things.  I hope it doesn't confuse people in the future when they look at computeIngressDegradedCondition and there is no information in that function that points to why unmanaged CRDs make it degraded.

Reply:
I don't see the departure.  We now combine multiple inputs to combine the "Degraded" status condition, same as we do for several other status conditions.  We set the status condition's `status` field to reflect these inputs, and we still use `mergeConditions` to update the complete set of status conditions (included "Degraded" and others) and only update a status condition's `lastTransitionTime` if its `state` has changed.  

If you have any suggestions for code comments that could clarify the logic, we can add those (perhaps in a follow-up PR).  The updated logic looks fine to me (but for https://github.com/openshift/cluster-ingress-operator/pull/1205#discussion_r2032181747), so I would like for us to proceed with this design.